centreon/src/Core/AdditionalConnectorConfiguration/Application/UseCase/AddAcc/AddAcc.php

```diff
@@ -0,0 +1,191 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\AdditionalConnectorConfiguration\Application\UseCase\AddAcc;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Core\AdditionalConnectorConfiguration\Application\Exception\AccException;
+use Core\AdditionalConnectorConfiguration\Application\Factory\AccFactory;
+use Core\AdditionalConnectorConfiguration\Application\Repository\ReadAccRepositoryInterface;
+use Core\AdditionalConnectorConfiguration\Application\Repository\WriteAccRepositoryInterface;
+use Core\AdditionalConnectorConfiguration\Application\Repository\WriteVaultAccRepositoryInterface;
+use Core\AdditionalConnectorConfiguration\Domain\Model\Acc;
+use Core\AdditionalConnectorConfiguration\Domain\Model\NewAcc;
+use Core\AdditionalConnectorConfiguration\Domain\Model\Poller;
+use Core\AdditionalConnectorConfiguration\Domain\Model\Type;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Common\Infrastructure\FeatureFlags;
+
+final class AddAcc
+{
+    use LoggerTrait;
+
+    /** @var WriteVaultAccRepositoryInterface[] */
+    private array $writeVaultAccRepositories = [];
+
+    /**
+     * @param ReadAccRepositoryInterface $readAccRepository
+     * @param WriteAccRepositoryInterface $writeAccRepository
+     * @param Validator $validator
+     * @param AccFactory $factory
+     * @param DataStorageEngineInterface $dataStorageEngine
+     * @param ContactInterface $user
+     * @param FeatureFlags $flags
+     * @param \Traversable<WriteVaultAccRepositoryInterface> $writeVaultAccRepositories
+     */
+    public function __construct(
+        private readonly ReadAccRepositoryInterface $readAccRepository,
+        private readonly WriteAccRepositoryInterface $writeAccRepository,
+        private readonly Validator $validator,
+        private readonly AccFactory $factory,
+        private readonly DataStorageEngineInterface $dataStorageEngine,
+        private readonly ContactInterface $user,
+        private readonly FeatureFlags $flags,
+        \Traversable $writeVaultAccRepositories,
+    ) {
+        $this->writeVaultAccRepositories = iterator_to_array($writeVaultAccRepositories);
+    }
+
+    public function __invoke(
+        AddAccRequest $request,
+        AddAccPresenterInterface $presenter
+    ): void {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_ACC_RW)) {
```

Comment:
according to the constant in contact this will not work (I guess we remove the CONFIGURATION before RW ^^)